l10n_ch_pain_credit_transfer/models/account_payment_order.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# copyright 2018 Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, api
+
+
+class AccountPaymentOrder(models.Model):
+    _inherit = 'account.payment.order'
+
+    @api.model
+    def _get_line_key(self, line):
+        return super(AccountPaymentOrder, self)._get_line_key(line) + (
+            line.currency_id.name,
+        )
+
+    @api.model
+    def generate_start_payment_info_block(
+            self, parent_node, payment_info_ident,
+            priority, local_instrument, category_purpose, sequence_type,
+            requested_date, eval_ctx, gen_args):
```

Comment:
Can you add a docstring to explain the change here where you prepare a more complex value for start_payment in order to differenciate from other blocks.

Also, I'm not sure of which part of the payment_info_ident attribute is going to be the currency itself.